pkgs/applications/misc/organicmaps/default.nix

```diff
@@ -0,0 +1,94 @@
+{ lib
+, mkDerivation
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, ninja
+, pkg-config
+, which
+, python3
+, makeWrapper
+, qtbase
+, qtsvg
+, libGLU
+, zlib
+}:
+
+mkDerivation rec {
+  pname = "organicmaps";
+  version = "2021-05-15";
+
+  src = fetchFromGitHub {
+    owner = "organicmaps";
+    repo = "organicmaps";
+    rev = version;
+    sha256 = "sha256-g5AV72JZjgcWKTBkve1dAi9W9ocdeWAROy+HCrsILLE=";
+    fetchSubmodules = true;
```

Comment:
All the 3party things should be devondered because upstream is most likely behind in security patches.

Reply:
This will be difficult. The project is configured to build those dependencies from source and doesn't detect the system ones. I'd have to patch it upstream, but I'm not very knowledgeable about cmake, so that'd take a while.

Is there a policy for vendored dependencies? There definitely are other packages where we allow it (eg. most of the rust and go based projects).

Reply:
Of course I could also substitute the bundled sources with the `.src`s of the ones we have in nixpkgs but that sounds like a massive hack

Reply:
> Is there a policy for vendored dependencies?

Not really it but general distro rule is to not do it unless it is necessary.

> There definitely are other packages where we allow it (eg. most of the rust and go based projects).

Those languages are harder or impossible to devendor but c++/cmake projects are usually pretty easy and upstream supports it or we can re-use patches from other distros.

Reply:
[No patches from other distros, unfortunately :-/](https://repology.org/projects/?search=organicmaps). I'll ask the other people in https://github.com/organicmaps/organicmaps/issues/395 if anyone can help, and otherwise I'll postpone this until I have time to learn how to patch their cmake scripts.

Reply:
Freetype and icu are now devendored! https://github.com/organicmaps/organicmaps/issues/395#issuecomment-1019482630